### PR TITLE
[Executor] Enrich error message for flex flow

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Currently for exceptions failed in flex flow's user code. The stack trace will lost and entry name is incorrect.
This PR added trace back to exception and refined entry name in error message.

Before:
image

After:
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/executor/_script_executor.py [hunk 1]
```diff
@@ -427,6 +429,8 @@ def _parse_entry_func(self):
 
     def _initialize_function(self):
         func = self._parse_entry_func()
+        # persist original function for detailed error message
+        self._original_function = func
```
Comment (thy09) [inline src/promptflow-core/promptflow/executor/_script_executor.py:433]: What if the function is already wrapped by @trace?

Comment (D-W-) [reply src/promptflow-core/promptflow/executor/_script_executor.py]: Updated to get from traced function

File: src/promptflow-core/promptflow/executor/_script_executor.py [hunk 2]
```diff
@@ -72,6 +72,15 @@ def __init__(
         self._message_format = MessageFormatType.BASIC
         self._multimedia_processor = BasicMultimediaProcessor()
 
+    def _get_func_name(self):
+        try:
+            original_func = getattr(self._func, "__original_function")
```
Comment (crazygao) [inline src/promptflow-core/promptflow/executor/_script_executor.py:77]: async function?

Comment (thy09) [reply src/promptflow-core/promptflow/executor/_script_executor.py:77]: I guess we can store the name once we initialize this executor.

Comment (D-W-) [reply src/promptflow-core/promptflow/executor/_script_executor.py:77]: For async function, seems `self._func` also store a sync function? 
image

Comment (D-W-) [reply src/promptflow-core/promptflow/executor/_script_executor.py]: stored to executor's field

**Review by 0mza987 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**